jormungandr/src/fragment/pool.rs

```diff
@@ -123,15 +134,17 @@ impl Pools {
                 continue;
             }
 
-            if let Some(mut persistent_log) = self.persistent_log.as_mut() {
+            if let Some(persistent_log) = self.persistent_log.as_mut() {
                 let entry = PersistentFragmentLog {
                     time: SecondsSinceUnixEpoch::now(),
                     fragment: fragment.clone(),
                 };
                 // this must be sufficient: the PersistentFragmentLog format is using byte array
                 // for serialization so we do not expect any problems during deserialization
                 let codec = bincode::DefaultOptions::new().with_fixint_encoding();
-                if let Err(err) = codec.serialize_into(&mut persistent_log, &entry) {
```

Comment:
That's why it was slow: we gave the `File` writer to bincode to do lots of small writes.